ui/components/logs/ui/code-editor.tsx

```
@@ -0,0 +1,492 @@
+						editorContainer.current.style.overflowY = "auto";
+						editorContainer.current.style.maxHeight = `${props.maxHeight || contentHeight}px`;
+					}
+				});
+			}
+		}
+		if (props.customLanguage) {
+			props.customLanguage.register(monaco);
+			editor.onDidChangeModelContent(() => {
+				const model = editor.getModel();
+				if (!model) return;
+				const markers = props.customLanguage!.validate(monaco, model);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)